extensions/opentelemetry/runtime/src/main/java/io/quarkus/opentelemetry/runtime/OpenTelemetryUtil.java

```diff
@@ -54,22 +59,41 @@ public static Map<String, String> convertKeyValueListToMap(List<String> headers)
      * @param vertxContext vertx context
      */
     public static void setMDCData(Context context, io.vertx.core.Context vertxContext) {
+        setMDCData(getSpanData(context), vertxContext);
+    }
+
+    public static void setMDCData(Map<String, String> spanData, io.vertx.core.Context vertxContext) {
+        if (spanData != null && !spanData.isEmpty()) {
+            spanData.entrySet().stream()
+                    .filter(entry -> SPAN_DATA_KEYS.contains(entry.getKey())) // filter out unknown keys
+                    .forEach(entry -> VertxMDC.INSTANCE.put(entry.getKey(), entry.getValue(), vertxContext));
+        }
```

Comment:
We generally try pretty hard to not have lambdas in runtime code

Reply:
Sorry...